.github/workflows/pr-realm-js.yml

```diff
@@ -0,0 +1,68 @@
+name: pr-realm-js
+on: pull_request
+
+jobs:
+  build-prebuild:
+    name: Build for Node.js on ${{ matrix.variant.os }} ${{ matrix.variant.arch }}
+    runs-on: ${{ matrix.variant.runner }}
+    strategy:
+      #fail-fast: true
+      matrix:
+        variant:
+          - { os: linux, runner: ubuntu-latest, arch: x64 }
+          #- { os: windows, runner: windows-latest, arch: x64 }
+          #- { os: windows, runner: windows-latest, arch: ia32 }
+          #- { os: darwin, runner: macos-latest, arch: x64 }
+          #- { os: ios, runner: macos-latest, arch: x64 }
+          - { os: android, runner: ubuntu-latest, arch: x86_64 }
+          - { os: android, runner: ubuntu-latest, arch: armeabi-v7a }
+          - { os: android, runner: ubuntu-latest, arch: arm64-v8a }
+          - { os: android, runner: ubuntu-latest, arch: x86 }
+    steps:
+      # This will retrieve the realm-js github repository
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          submodules: "recursive"
+
+      # This sets node up, and can also pick the version if necessary
+      - name: Setup node version
+        uses: actions/setup-node@v1
+
+      - name: Cache prebuild artifacts
+        id: cache-prebuild
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-prebuild-artifacts
+        with:
+          path: prebuilds
+          key: ${{ runner.os }}-${{matrix.variant.os}}-${{matrix.variant.arch}}-prebuild-${{hashFiles('./src/**', 'vendor/**')}}
+
+      - name: MSVC Setup
+        if: steps.cache-prebuild.outputs.cache-hit != 'true'
+        uses: ilammy/msvc-dev-cmd@v1
+
+      - name: Windows Environment setup
+        if: ${{ (matrix.variant.os == 'windows') && (steps.cache-prebuild.outputs.cache-hit != 'true') }}
+        run: |
+          Invoke-WebRequest -Uri https://static.realm.io/downloads/vcpkg.zip -OutFile C:\vcpkg.zip
+          Expand-Archive -Path C:\vcpkg.zip -DestinationPath C:\
+        shell: powershell
+
+      - name: Linux Environment setup
+        if: ${{ (matrix.variant.os == 'linux') && (steps.cache-prebuild.outputs.cache-hit != 'true') }}
+        run: |
+          sudo apt-get install ninja-build
+
+      # Install node dependencies and build the c++ library
+      - name: Build
+        if: steps.cache-prebuild.outputs.cache-hit != 'true'
+        run: |
+          npm ci --ignore-scripts
+          npm run build-${{ matrix.variant.os }} -- --arch=${{matrix.variant.arch}}
```

Comment:
i think i'd like the build script to be of the naming convention `build:{os}`. so for example: 
```
"build:android": ...
"build:linux": ...
etc...
```